### PR TITLE
Disable LF2

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**

### CODE REVIEW

**Review by simonmaxen-da — APPROVED** [review]
LGTM

File: compiler/damlc/tests/daml-test-files/Daml3ScriptPackageVetting.daml [hunk 1]
```diff
@@ -99,10 +99,10 @@ listPackagesIsCorrect = script do
   assertAllPackages
 
   -- Check default packages count (as packageA and packageB previously disabled)
-  -- 12 packages in daml-prim (11 specific modules + prim itself)
-  -- 14 packages in daml-stdlib (13 specific modules + stdlib itself)
-  -- 2 packages in ghc-stdlib
+  -- 6 packages in daml-prim (11 specific modules + prim itself)
+  -- 6 packages in daml-stdlib (13 specific modules + stdlib itself)
+  -- 1 packages in ghc-stdlib
```
Comment (remyhaemmerle-da) [inline compiler/damlc/tests/daml-test-files/Daml3ScriptPackageVetting.daml:104]:
```suggestion
  -- 6 packages in daml-prim (5 specific modules + prim itself)
  -- 6 packages in daml-stdlib (5 specific modules + stdlib itself)
  -- 1 packages in ghc-stdlib
```

File: compiler/damlc/tests/BUILD.bazel [hunk 2]
```diff
@@ -210,12 +210,12 @@ da_haskell_library(
...
 
 da_haskell_test(
-    name = "integration-v2dev-script2",
+    name = "integration-v1dev-script2",
     size = "large",
     srcs = ["src/DA/Test/DamlcIntegrationMain.hs"],
     args = [
         "--daml-lf-version",
-        "2.dev",  # script2 only supports 2.x
+        "1.dev",  # script2 only supports 2.x
```
Comment (paulbrauner-da) [inline compiler/damlc/tests/BUILD.bazel:218]:
```suggestion
        "1.dev",
```

File: daml-script/daml3/Daml/Script/Questions/Submit/Error.daml
Comment (paulbrauner-da) [inline daml-script/daml3/Daml/Script/Questions/Submit/Error.daml:1]: Is this change a consequence of not supporting lf2?

**Review by paulbrauner-da — APPROVED** [review]
LGTM but curious as to why the Error.daml change is bundled with this PR.

File: daml-script/daml3/Daml/Script/Questions/Submit/Error.daml
Comment (remyhaemmerle-da) [reply daml-script/daml3/Daml/Script/Questions/Submit/Error.daml:1]: Yes. we do not have `templateTypeRepToText` in 1.15

Comment (remyhaemmerle-da) [reply daml-script/daml3/Daml/Script/Questions/Submit/Error.daml:1]: This is basically a backport of https://github.com/digital-asset/daml/pull/18298

**Pull request merged**